docker/scripts/cyber_start.sh

```diff
@@ -15,265 +15,509 @@
+                    "and nvidia-container-toolkit as described by: "
+            warning "  ${nv_docker_doc}"
+        fi
+    fi
+}
+
+function setup_devices_and_mount_volumes() {
+    local __retval="$1"
 
-    APOLLO_CYBER="apollo_cyber_${USER}"
-    docker ps -a --format "{{.Names}}" | grep "$APOLLO_CYBER" 1>/dev/null
-    if [ $? == 0 ]; then
-        docker stop $APOLLO_CYBER 1>/dev/null
-        docker rm -v -f $APOLLO_CYBER 1>/dev/null
+    if [[ "${HOST_ARCH}" == "${TARGET_ARCH}" ]]; then
+        source "${APOLLO_ROOT_DIR}/scripts/apollo_base.sh" CYBER_ONLY
+        setup_device
     fi
 
-    local display=""
-    if [[ -z ${DISPLAY} ]];then
+    local volumes
+    volumes="-v ${APOLLO_ROOT_DIR}:/apollo"
+
+    local kernel="$(uname -s)"
+    if [[ "${kernel}" != "Linux" ]]; then
+        warning "Running Apollo cyber container on ${kernel} is UNTESTED, exiting..."
+        exit 1
+    fi
+
+    local os_release="$(lsb_release -rs)"
+    case "${os_release}" in
+        14.04)
+            warning "[Deprecated] Support for Ubuntu 14.04 will be removed" \
+                    "in the near future. Please upgrade to ubuntu 18.04+."
+            ;;
+        16.04|18.04|20.04|*)
+            ## Question(storypku): Any special considerations here ?
+            if [[ "${HOST_ARCH}" == "${TARGET_ARCH}" ]]; then
+                volumes="${volumes} -v /dev:/dev"
+            fi
+            ;;
+    esac
+
+    volumes="${volumes} -v /etc/localtime:/etc/localtime:ro"
+    # volumes="${volumes} -v /usr/src:/usr/src"
+    if [[ "${HOST_ARCH}" == "${TARGET_ARCH}" ]]; then
+        volumes="${volumes} -v /dev/null:/dev/raw1394 \
+                            -v /media:/media \
+                            -v /tmp/.X11-unix:/tmp/.X11-unix:rw \
+                            -v /lib/modules:/lib/modules \
+                    "
+
+    fi
+    volumes="$(tr -s " " <<< "${volumes}")"
+    eval "${__retval}='${volumes}'"
+}
+
+function determine_display() {
+    local display
+    # read from env
+    if [[ -z "${DISPLAY}" ]]; then
         display=":0"
     else
         display="${DISPLAY}"
     fi
+    echo "${display}"
+}
 
-    setup_device
+function remove_existing_cyber_container() {
+    if docker ps -a --format '{{.Names}}' | grep -q "${CYBER_CONTAINER}"; then
+        info "Removing existing cyber container ${CYBER_CONTAINER}"
+        docker stop "${CYBER_CONTAINER}" >/dev/null
+        docker rm -v -f "${CYBER_CONTAINER}" >/dev/null
+    fi
+}
 
-    USER_ID=$(id -u)
-    GRP=$(id -g -n)
-    GRP_ID=$(id -g)
-    LOCAL_HOST=`hostname`
-    DOCKER_HOME="/home/$USER"
-    if [ "$USER" == "root" ];then
-        DOCKER_HOME="/root"
+function docker_pull_if_needed() {
+    local image="$1"
+    local use_local_image="$2"
+    if [[ ${use_local_image} -eq 1 ]]; then
+        info "Start cyber container based on local image: ${image}"
+    elif docker images -a --format '{{.Repository}}:{{.Tag}}' \
```

Comment:
Only pull image if not found locally. If `remote/repo:latest` is wanted, users have to `docker rmi` the local image manually.